.github/workflows/publish.yml

```
@@ -0,0 +1,112 @@
+          tags: |
+            ${{ vars.DOCKERHUB_USERNAME }}/${{ vars.IMAGE_NAME }}:${{ env.TOR_VERSION }}
+            ghcr.io/${{ github.repository_owner }}/${{ vars.IMAGE_NAME }}:${{ env.TOR_VERSION }}
+            ${{ vars.DOCKERHUB_USERNAME }}/${{ vars.IMAGE_NAME }}:latest
+            ghcr.io/${{ github.repository_owner }}/${{ vars.IMAGE_NAME }}:latest
+          platforms: linux/amd64
+          
+      - name: Attest DockerHub
+        uses: actions/attest-build-provenance@v4
+        with:
+          subject-name: docker.io/${{ vars.DOCKERHUB_USERNAME }}/${{ vars.IMAGE_NAME }}
+          subject-digest: ${{ steps.push.outputs.digest }}
```

actionlint (1.7.12)

[error] property "push" is not defined in object type {determine-ref: {conclusion: string; outcome: string; outputs: {string => string}}; torver: {conclusion: string; outcome: string; outputs: {string => string}}}

(expression)